packages/console/src/pages/OneTimeTokenLanding/index.tsx

```diff
@@ -0,0 +1,74 @@
+import { useLogto } from '@logto/react';
+import { FirstScreen } from '@logto/schemas';
+import { yes } from '@silverhand/essentials';
+import { useContext, useEffect } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+
+import AppLoading from '@/components/AppLoading';
+import { TenantsContext } from '@/contexts/TenantsProvider';
+import useRedirectUri from '@/hooks/use-redirect-uri';
+
+enum OneTimeTokenLandingSearchParams {
+  OneTimeToken = 'one_time_token',
+  Email = 'email',
+  IsNewUser = 'is_new_user',
+}
+
+/** The one-time token landing page for sign-in with one-time tokens. */
+function OneTimeTokenLanding() {
+  const navigate = useNavigate();
+  const { isAuthenticated, signIn, isLoading } = useLogto();
+  const [searchParams] = useSearchParams();
+  const { navigateTenant, currentTenantId } = useContext(TenantsContext);
+  const redirectUri = useRedirectUri();
+
+  useEffect(() => {
+    if (isAuthenticated) {
+      // If we have a current tenant, navigate to it
+      if (currentTenantId) {
+        navigateTenant(currentTenantId);
+      } else {
+        // Otherwise navigate to root, which will handle tenant selection
+        navigate('/', { replace: true });
+      }
+    }
+  }, [isAuthenticated, navigate, currentTenantId, navigateTenant]);
+
+  useEffect(() => {
+    const oneTimeToken = searchParams.get(OneTimeTokenLandingSearchParams.OneTimeToken);
+    const email = searchParams.get(OneTimeTokenLandingSearchParams.Email);
+
+    const isNewUser = yes(searchParams.get(OneTimeTokenLandingSearchParams.IsNewUser));
+
+    if (!oneTimeToken || !email) {
+      navigate('/', { replace: true });
+      return;
+    }
+
+    if (isLoading || isAuthenticated) {
+      return;
+    }
+
+    const handleSignIn = async () => {
+      try {
+        await signIn({
+          redirectUri,
+          clearTokens: false,
+          firstScreen: isNewUser ? FirstScreen.Register : FirstScreen.SignIn,
```

Comment:
This seems unnecessary to me. You don't have to distinguish register or sign-in when using one-time token. The token authentication flow will firstly try to register and if it fails, it updates the interaction event to `sign-in` and triggers a sign-in attemp instead.